pkg/controllers/state/suite_test.go

```diff
@@ -1431,3 +1433,132 @@ func ExpectStateNodeNotFoundForMachine(machine *v1alpha5.Machine) *state.Node {
 	ExpectWithOffset(1, ret).To(BeNil())
 	return ret
 }
+
+var _ = Describe("DaemonSet Controller", func() {
```

Comment:
Can we add a test for if the daemonset is updated and there is a newer pod that we should represent?